src/odemis/util/peak.py

```diff
@@ -264,8 +265,11 @@ def _DoFit(self, future, spectrum, wavelength, type='gaussian'):
                     raise CancelledError()
 
                 try:
-                    # => in scipy 0.17, curve_fit() supports the 'bounds' parameter
-                    params, _ = curve_fit(FitFunction, wavelength, spectrum, p0=fit_list)
+                    with warnings.catch_warnings():
+                        # Hide scipy/optimize/minpack.py:690: OptimizeWarning: Covariance of the parameters could not be estimated
+                        warnings.filterwarnings("ignore", "", OptimizeWarning)
+                        # => in scipy 0.17, curve_fit() supports the 'bounds' parameter
+                        params, _ = curve_fit(FitFunction, wavelength, spectrum, p0=fit_list)
```

Comment:
I don't completely understand the comment. We are using scipy 0.17 correct? So is the warning not happening in a newer version of scipy anymore? Will it be replaced somehow? Is it actually a TODO? scipy 0.17 supports bounds parameter, but you don't use/want to use it?

Reply:
To be honest, I was also a little confused by that comment, so I just left it as-is. That said, I think it's me who added it a long time ago. So, I've looked again at what it should mean, and now updated it.